src/integration/resources/docker/dockerexternal/etcd.go

```diff
@@ -164,6 +165,13 @@ func (c *EtcdNode) Setup(ctx context.Context) (closeErr error) {
 		return fmt.Errorf("starting etcd container: %w", err)
 	}
 
+	defer func() {
```

Comment:
This wasn't cleaning up properly before, leaving abandoned containers if startup failed here.